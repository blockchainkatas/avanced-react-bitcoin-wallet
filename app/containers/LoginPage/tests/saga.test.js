/**
 * Tests for LoginPage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { CHECK_TOKEN } from 'containers/App/constants';
import { tokenValidated, checkTokenError } from 'containers/App/actions';

import checkTokenWatcher, { checkToken } from '../saga';

const token = '14c71464e5d0323691q7c314b42eg04';

/* eslint-disable redux-saga/yield-effects */
describe('checkToken Saga', () => {
  let checkTokenGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    checkTokenGenerator = checkToken();

    const selectDescriptor = checkTokenGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = checkTokenGenerator.next(token).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the tokenValidated action if it requests the data successfully', () => {
    const response = {
      wallet_names: [
        '58A884F8A8A47869138798',
        '58A885479E786982212958',
      ],
    };
    const putDescriptor = checkTokenGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(tokenValidated(response, token)));
  });

  it('should call the checkTokenError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = checkTokenGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(checkTokenError(response)));
  });
});

describe('checkTokenWatcherSage Saga', () => {
  const checkTokenWatcherSage = checkTokenWatcher();

  it('should start task to watch for CHECK_TOKEN action', () => {
    const takeLatestDescriptor = checkTokenWatcherSage.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(CHECK_TOKEN, checkToken));
  });
});
