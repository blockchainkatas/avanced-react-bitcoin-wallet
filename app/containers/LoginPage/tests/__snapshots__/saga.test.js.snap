// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkToken Saga should call the checkTokenError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`checkToken Saga should call the checkTokenError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.blockcypher.com/v1/bcy/test/wallets?token=14c71464e5d0323691q7c314b42eg04",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`checkToken Saga should dispatch the tokenValidated action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`checkToken Saga should dispatch the tokenValidated action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.blockcypher.com/v1/bcy/test/wallets?token=14c71464e5d0323691q7c314b42eg04",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
